docs/source/pages/tutorials/commit_finder.rst

```diff
@@ -0,0 +1,162 @@
+
+* A commit at a Git repository that corresponds to the artifact (``mcn_version_control_system_1``)
+* The build tool ``pip`` used in the build scripts (``mcn_build_script_1``)
+* GitHub Actions workflow to build the package (``mcn_build_service_1``)
+* GitHub Actions workflow to deploy and publish the package (``mcn_build_as_code_1``)
+
+Now we should run the next analysis, and then open the new report.
+
+.. code-block:: shell
+
+    ./run_macaron.sh analyze -purl pkg:pypi/arrow@0.15.0 --skip-deps
+    open output/reports/pypi/arrow/arrow.html
+
+.. _fig_arrow_0.15.0_top:
+
+.. figure:: ../../_static/images/tutorial_arrow_0.15.0_report_top.png
+   :alt: HTML report for ``arrow 0.15.0``, summary
+   :align: center
+
+.. _fig_arrow_0.15.0:
+
+.. figure:: ../../_static/images/tutorial_arrow_0.15.0_report.png
+   :alt: HTML report for ``arrow 0.15.0``, checks
+   :align: center
+
+In the second report for Arrow, we can see that Macaron has returned different results. Starting with the ``Target Information`` section we can see that the repository for this older artifact is not the same as the current one: ``https://github.com/crsmithdev/arrow`` instead of ``https://github.com/arrow-py/arrow``. In the checks section, we can see that two of the four checks that passed for the previous version, did not pass for this earlier version. Checks ``mcn_build_service_1`` and ``mcn_build_as_code_1`` failed, indicating that the older artifact did not have a GitHub Actions workflow setup to build or publish the package. In this way Macaron has demonstrated the usefulness of being able to analyze a repository at multiple stages, thereby allowing for a more accurate analysis when investigating artifacts that are, or use, outdated libraries.
+
+*****************************
+Run ``verify-policy`` command
+*****************************
+
+Another feature of Macaron is policy verification. This allows Macaron to report on whether an artifact meets the security requirements specified by the user. Policies are written using `Soufflé Datalog <https://souffle-lang.github.io/index.html>`_ , a language similar to SQL. Results collected by the ``analyze`` command can be checked via declarative queries in the created policy, which Macaron can then automatically check.
+
+The security requirement chosen for this tutorial reflects the difference between the two versions in the previous section. That is, we want to ensure that the artifact has a valid build service. If we refer back to :ref:`step <fig_arrow_0.15.0>` and :ref:`step <fig_arrow_1.3.0>`, we can see that the relevant check ID of the difference between the two versions is ``mcn_build_as_code_1``. To include this in a policy we create the following:
+
+.. code-block:: c++
+
+    #include "prelude.dl"
+
+    Policy("has-hosted-build", component_id, "Require a hosted build and publishing service.") :-
+        check_passed(component_id, "mcn_build_as_code_1").
```

Comment:
Should the other failed check be added here as well?

Reply:
I have tried to clarify that in the description.

Reply:
See commit [4f11927](https://github.com/oracle/macaron/pull/606/commits/4f11927a8dd7d67b9aa1c61c31c980bba03f24fa)